### PR TITLE
[DataGrid] Add dark mode scrollbar

### LINKED ISSUE #260: [DataGrid] Dark mode scrollbar
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

image

## Expected Behavior 🤔

No white area

## Steps to Reproduce 🕹

1. Open https://master--material-ui-x.netlify.app/components/data-grid/#commercial-version
2. Turn dark mode

## Context 🔦

Most traders display the grid on a dark background, take https://openfin.co/ or https://perspective.finos.org/ as an example.

## Your Environment 🌎

| Tech          | Version |
| ------------- | ------- |
| Material-UI X | v0.1.67  |

### PR DESCRIPTION
**Pull request by dtassone**
Closes #260

### CODE REVIEW

File: packages/grid/x-grid-modules/src/components/styled-wrappers/GridRootStyles.ts [hunk 1]
```diff
@@ -203,6 +203,22 @@ export const useStyles = makeStyles(
...
+          width: 'initial',
+          height: 'initial',
+        },
+        '& *::-webkit-scrollbar-thumb': {
+          borderRadius: 10,
+          backgroundColor: borderColor,
+        },
+      };
+    }
+    return gridStyle;
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/components/styled-wrappers/GridRootStyles.ts:221]:
```suggestion
    if (theme.palette.type === 'dark') {
      // Values coming from mac OS.
      const track = '#202022';
      const thumb = '#585859';
      const active = '#838384';

      // We style the scroll bar for dark mode.
      gridStyle.root = {
        ...gridStyle.root,
        scrollbarColor: `${thumb} ${track}`,
        '& *::-webkit-scrollbar': {
          backgroundColor: track,
        },
        '& *::-webkit-scrollbar-thumb': {
          borderRadius: 8,
          backgroundColor: thumb,
          minHeight: 24,
          border: `3px solid ${track}`,
        },
        '& *::-webkit-scrollbar-thumb:focus': {
          backgroundColor: active,
        },
        '& *::-webkit-scrollbar-thumb:active': {
          backgroundColor: active,
        },
        '& *::-webkit-scrollbar-corner': {
          backgroundColor: track,
        },
      };
    }
    return gridStyle;
```

File: packages/grid/x-grid-modules/src/components/styled-wrappers/GridRootStyles.ts [hunk 2]
```diff
@@ -2,12 +2,12 @@ import { darken, fade, lighten, makeStyles, Theme } from '@material-ui/core/styl
...
-        ? lighten(fade(theme.palette.divider, 1), 0.88)
-        : darken(fade(theme.palette.divider, 1), 0.68);
-
-    return {
+    const isLightTheme = theme.palette.type === 'light';
+    const isDarkTheme = !isLightTheme;
+    const borderColor = isLightTheme
+      ? lighten(fade(theme.palette.divider, 1), 0.88)
+      : darken(fade(theme.palette.divider, 1), 0.68);
+    const gridStyle: { root: any } = {
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/components/styled-wrappers/GridRootStyles.ts:10]:
```suggestion
    const borderColor =
    theme.palette.type === 'light'
      ? lighten(fade(theme.palette.divider, 1), 0.88)
      : darken(fade(theme.palette.divider, 1), 0.68);

    const gridStyle: { root: any } = {
```

**Review by oliviertassinari — COMMENTED** [review]
I have proposed an improved version, it handles:

- Add support for Firefox
Capture d’écran 2020-09-14 à 13 26 31

- Add support for the corner color:
Capture d’écran 2020-09-14 à 13 26 58

- Improve the contrast radio: https://webaim.org/resources/contrastchecker/?fcolor=404040&bcolor=191919.
- Add a min size for the thumb, as the default version has
- Add an "active" color when dragging with even more contrast.

Final UX:

after

**Comment by oliviertassinari** [comment]
I'm testing on Windows now.

**Comment by oliviertassinari** [comment]
> I'm testing on Windows now.

All good on Windows.

For those who wonder, the solution is coming from StackOverflow, not a question, [from their source](https://github.com/StackExchange/Stacks/blob/a7d3c05ef712b909c75e4ac4e393ab683433f5df/lib/css/exports/_stacks-constants-helpers.less#L102) 😄.

cc @mui-org/core-team What if we applied the customization of the scrollbar to our CssBasline component? Or at least to the documentation styles?

Capture d’écran 2020-09-14 à 13 38 45

**Comment by eps1lon** [comment]
Let's discuss this in a separate issue so that we don't make this PR too noisy.

**Comment by oliviertassinari** [comment]
Sure, done: https://github.com/mui-org/material-ui/issues/22594

**Review by oliviertassinari — APPROVED** [review]

**Pull request merged**